lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -756,6 +780,21 @@ void CBoostedTreeFactory::initializeUnsetRegularizationHyperparameters(core::CDa
                   << core::CContainerPrinter::print(gainAndTotalCurvaturePerNode));
     });
 
+    // initialize unset regularization hyperparameters with meaningful values
+    if (hyperparameters.earlyStoppingEnabled()) {
```

Comment:
I'm not sure about this being conditional on enabling early stopping. It seems weird to me to change this behaviour based on this flag. My feeling is it is probably better to use the true minimum we consider rather than zero anyway.

Reply:
This turned out to be due to handling of max number trees.